Carrito.js

```diff
@@ -0,0 +1,193 @@
+import Notificaciones from "./Notificaciones.js";
+
+class Carrito {
+    constructor() {
+        this.productos = JSON.parse(localStorage.getItem("carrito")) || [];
+        this.notis = new Notificaciones(this);
+        this.formateador = new Intl.NumberFormat("es-MX", { style: "currency", currency: "MXN" });
+    }
+
+    guardarCarrito() {
+        localStorage.setItem("carrito", JSON.stringify(this.productos));
+    }
+
+    validarProducto(nombreProducto, precioProducto) {
+        if (!nombreProducto || typeof nombreProducto !== "string" || nombreProducto.trim() === "") {
+            console.error("Nombre de producto inválido:", nombreProducto);
+            return false;
+        }
+
+        if (isNaN(precioProducto) || precioProducto <= 0) {
+            console.error("Precio de producto inválido:", precioProducto);
+            return false;
+        }
+
+        return true;
+    }
+
+    agregarAlCarrito(nombreProducto, precioProducto) {
+        if (!this.validarProducto(nombreProducto, precioProducto)) {
+            this.notis.mostrarNotificacion("Datos inválidos al intentar agregar un producto.", "error");
+            return;
+        }
+
+        const productoExistente = this.productos.find((producto) => producto.nombre === nombreProducto);
+
+        if (productoExistente) {
+            productoExistente.cantidad += 1;
+            this.notis.mostrarNotificacion(`Cantidad aumentada para el producto: ${nombreProducto}`, "info");
+        } else {
+            this.productos.push({ nombre: nombreProducto, precio: Number(precioProducto), cantidad: 1 });
+            this.notis.mostrarNotificacion(`Producto agregado: ${nombreProducto}`, "success");
+        }
+
+        this.guardarCarrito();
+        this.actualizarCarrito();
+    }
+
+    eliminarDelCarrito(nombreProducto) {
+        const productoExistente = this.productos.find((producto) => producto.nombre === nombreProducto);
+
+        if (productoExistente) {
+            this.productos = this.productos.filter((producto) => producto.nombre !== nombreProducto);
```

Comment:
Mantente consistente asignando el resultado del filtro a una nueva variable primero para asegurar que las notificaciones se muestren correctamente antes de modificar this.productos.
`const nuevoListado = this.productos.filter((producto) => producto.nombre !== nombreProducto);`